atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbReadTable.java

```diff
@@ -105,7 +105,7 @@ public DbReadTable(ConnectionSupplier conns, DbQueryFactory queryFactory) {
         FullQuery query = queryFactory.getRowsColumnRangeCountsQuery(rows, ts, columnRangeSelection);
         AgnosticLightResultSet results = conns.get()
                 .selectLightResultSetUnregisteredQuery(query.getQuery(), query.getArgs());
-        results.setFetchSize(Math.max(rows.size(), MAX_ROW_COLUMN_RANGES_FETCH_SIZE));
+        results.setFetchSize(Math.min(rows.size(), MAX_ROW_COLUMN_RANGES_FETCH_SIZE));
```

Comment:
If the result is empty then the fetch size probably shouldn't matter, but short-circuiting makes sense anyway, will update the PR.